Detectors/ITSMFT/ITS/RawPixelReaderWorkflow/src/RawPixelReaderSpec.cxx

```diff
@@ -0,0 +1,608 @@
+#include <vector>
+#include <TTree.h>
+#include <TFile.h>
+#include <TStopwatch.h>
+#include <string>
+#include "TTree.h"
+
+#include "Framework/ControlService.h"
+#include "ITSMFTReconstruction/ChipMappingITS.h"
+#include "ITSMFTReconstruction/GBTWord.h"
+#include "ITSMFTReconstruction/PayLoadCont.h"
+#include "ITSMFTReconstruction/PixelData.h"
+#include "DataFormatsITSMFT/ROFRecord.h"
+#include "ITSMFTReconstruction/RawPixelReader.h"
+#include "CommonDataFormat/InteractionRecord.h"
+#include "ITSRawWorkflow/RawPixelReaderSpec.h"
+#include "DetectorsBase/GeometryManager.h"
+#include <TCanvas.h>
+#include <iostream>
+#include <dirent.h>
+#include <stdio.h>
+#include <algorithm>
+#include <iterator>
+#include <thread>
+
+using namespace o2::framework;
+using namespace o2::itsmft;
+using namespace std;
+
+namespace o2
+{
+namespace its
+{
+
+void RawPixelReader::init(InitContext& ic)
+{
+  IndexPush = 0;
+
+  std::ifstream RunFileType("Config/RunType.dat");
+  RunFileType >> RunType;
+
+  LOG(DEBUG) << "OLD CONFIG: RunFileType = " << RunType;
+
+  if (RunType == 0) {
+    std::ifstream EventPush("Config/ConfigFakeRate.dat");
+    EventPush >> EventPerPush >> TrackError >> workdir;
+  }
+
+  if (RunType == 1) {
+    std::ifstream EventPush("Config/ConfigThreshold.dat");
+    EventPush >> EventPerPush >> TrackError >> workdir;
+  }
+
+  LOG(DEBUG) << "OLD CONFIG: EventPerPush = " << EventPerPush << "   TrackError = " << TrackError << "  workdir = " << workdir;
+  LOG(DEBUG) << "DONE Reset Histogram Decision";
+
+  o2::base::GeometryManager::loadGeometry();
+
+  FolderNames = GetFName(workdir);
+
+  cout << "NFolder = " << FolderNames.size() << endl;
+  for (int i = 0; i < FolderNames.size(); i++) {
+
+    cout << "FDN = " << FolderNames[i] << endl;
+
+    FileNames.push_back(GetFName(FolderNames[i]));
+
+    cout << "FDN File Size = " << FileNames[i].size() << endl;
+
+    for (int j = 0; j < FileNames[i].size(); j++) {
+
+      cout << "FDN File = " << FileNames[i][j] << endl;
+    }
+  }
+  for (int i = 0; i < NError; i++) {
+    Error[i] = 0;
+  }
+
+  //		EventPerPush = 3000;
+  EventRegistered = 0;
+  TotalPixelSize = 0;
+
+  //	GetFileName("infile");
+
+  //
+  o2::its::GeometryTGeo* geom = o2::its::GeometryTGeo::Instance();
+  geom->fillMatrixCache(o2::utils::bit2Mask(o2::TransformType::L2G));
+  const Int_t numOfChips = geom->getNumberOfChips();
+  LOG(DEBUG) << "numOfChips = " << numOfChips;
+  setNChips(numOfChips);
+  j = 0;
+  FileDone = 1;
+  PercentDone = 0.0;
+  FileRemain = 0;
+  TimeCounter = 0;
+  Printed = 0;
+  difference = 0;
+  differenceDecoder = 0;
+  NewFileInj = 1;
+  MaxPixelSize = 58700095;
+}
+
+void RawPixelReader::run(ProcessingContext& pc)
+{
+
+  cout << "------------------------------------------------------------------------------------------" << endl;
+  cout << "------------------------------------------------------------------------------------------" << endl;
+
```

Comment:
The code from here till L277 is extremely custom and almost incomprehensible. Could you comment it and put in a separate method?

Reply:
Comments added. I hope it will be clearer!